core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/LdapAuthorizationRule.scala

```diff
@@ -74,7 +74,28 @@ object LdapAuthorizationRule {
     override val name = Rule.Name("ldap_authorization")
   }
 
-  final case class Settings(ldap: LdapAuthorizationService,
-                            permittedGroups: UniqueNonEmptyList[Group],
-                            allLdapGroups: UniqueNonEmptyList[Group])
+  final case class Settings(ldap: LdapAuthorizationService, permittedGroups: GroupsLogic, allLdapGroups: UniqueNonEmptyList[Group])
```

Comment:
you add this to LdapAuthorizationRule. What about other authorization rules? I understand that this is not needed by the client now, but IMO we have to create a jira for `groups_and` support in other rules and add it to the backlog.

Reply:
The only one implementor of `BaseAuthorizationRule` is `ExternalAuthorizationRule` (not entirely sure why! I.e. JWT and ror_kbn_auth could be good candidate to join the band for example).

Will add the task for ExternalAuthorizationRule for now.

Reply:
the answer is rather simple - `LdapAuthorizationRule` and `ExternalAuthorizationRule` are authorization rules only. `JwtAuthRule` and `RorKbnAuthRule` are authorization and authentication rules at the same time (aka "auth rules"). Usage of the common base is not so obvious.

But the JWT based rules have a "groups" field in the configuration. So, they should support "groups_and" too. Will you add the proper task for them too?